hivemq-edge/src/frontend/src/components/rjsf/MqttTransformation/JsonSchemaBrowser.tsx

```diff
@@ -1,33 +1,47 @@
 import { type FC, useMemo } from 'react'
 import type { JSONSchema7 } from 'json-schema'
-import { List, ListProps } from '@chakra-ui/react'
+import { Heading, List, ListProps } from '@chakra-ui/react'
 
 import { getPropertyListFrom } from '@/components/rjsf/MqttTransformation/utils/json-schema.utils.ts'
 import PropertyItem from '@/components/rjsf/MqttTransformation/components/schema/PropertyItem.tsx'
 
 interface JsonSchemaBrowserProps extends ListProps {
   schema: JSONSchema7
   isDraggable?: boolean
+  hasExamples?: boolean
 }
 
-const JsonSchemaBrowser: FC<JsonSchemaBrowserProps> = ({ schema, isDraggable = false, ...props }) => {
+const JsonSchemaBrowser: FC<JsonSchemaBrowserProps> = ({
+  schema,
+  isDraggable = false,
+  hasExamples = false,
+  ...props
+}) => {
   const properties = useMemo(() => {
     return getPropertyListFrom(schema)
   }, [schema])
 
   return (
-    <List {...props}>
-      {properties.length > 0 &&
-        properties.map((property) => {
-          return (
-            <PropertyItem
-              key={[...property.path, property.title].join('-')}
-              property={property}
-              isDraggable={isDraggable}
-            />
-          )
-        })}
-    </List>
+    <>
+      {schema.title && (
+        <Heading as="h4" size="sm">
+          {schema.title}
+        </Heading>
+      )}
+      <List {...props}>
+        {properties.length > 0 &&
```

Comment:
genuine question: if `properties.length` exists, then at least we know `properties` is an array. I'm not convinced checking that there's at least 1 item in the `properties` array before doing `properties.map()` changes anything 🤷‍♂️. The `map()` should act like a `for` loop basically, so I assume you'd get the same result and no side effect if you removed the `properties.length > 0` test and `properties` was an empty array.

Reply:
Yes it does.